ckan/logic/action/get.py

```diff
@@ -1884,9 +1884,12 @@ def package_search(context, data_dict):
         include_drafts = asbool(data_dict.pop('include_drafts', False))
         data_dict.setdefault('fq', '')
         if not include_private:
-            data_dict['fq'] = '+capacity:public ' + data_dict['fq']
+            data_dict['fq'] = 'capacity:public' if not data_dict['fq'] \
+                else 'capacity:public +' + data_dict['fq']
         if include_drafts:
             data_dict['fq'] += ' +state:(active OR draft)'
+        if not include_drafts and not context.get('sysadmin'):
```

Comment:
It would be better to use `context['auth_user_obj']` instead of unneccessary `sysadmin` key

Reply:
Thanks, will do that instead.